x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -0,0 +1,159 @@
+
+    public static final String CONNECTOR_SYNC_JOB_INDEX_NAME = ConnectorTemplateRegistry.CONNECTOR_SYNC_JOBS_INDEX_NAME_PATTERN;
+
+    /**
+     * @param client A client for executing actions on the connectors sync jobs index.
+     */
+    public ConnectorSyncJobIndexService(Client client) {
+        this.clientWithOrigin = new OriginSettingClient(client, CONNECTORS_ORIGIN);
+    }
+
+    /**
+     * @param request   Request for creating a connector sync job.
+     * @param listener  Listener to respond to a successful response or an error.
+     */
+    public void createConnectorSyncJob(
+        PostConnectorSyncJobAction.Request request,
+        ActionListener<PostConnectorSyncJobAction.Response> listener
+    ) {
+        try {
+            getSyncJobConnectorInfo(request.getId(), listener.delegateFailure((l, connector) -> {
+                Instant now = Instant.now();
+                ConnectorSyncJobType jobType = Objects.requireNonNullElse(request.getJobType(), ConnectorSyncJob.DEFAULT_JOB_TYPE);
+                ConnectorSyncJobTriggerMethod triggerMethod = Objects.requireNonNullElse(
+                    request.getTriggerMethod(),
+                    ConnectorSyncJob.DEFAULT_TRIGGER_METHOD
+                );
+
+                try {
+                    String syncJobId = generateId();
+
+                    final IndexRequest indexRequest = new IndexRequest(CONNECTOR_SYNC_JOB_INDEX_NAME).id(syncJobId)
+                        .opType(DocWriteRequest.OpType.INDEX)
+                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                    ConnectorSyncJob syncJob = new ConnectorSyncJob.Builder().setId(syncJobId)
+                        .setJobType(jobType)
+                        .setTriggerMethod(triggerMethod)
+                        .setStatus(ConnectorSyncJob.DEFAULT_INITIAL_STATUS)
+                        .setConnector(connector)
+                        .setCreatedAt(now)
+                        .setLastSeen(now)
+                        .setTotalDocumentCount(ZERO)
+                        .setIndexedDocumentCount(ZERO)
+                        .setIndexedDocumentVolume(ZERO)
+                        .setDeletedDocumentCount(ZERO)
+                        .build();
+
+                    indexRequest.source(syncJob.toXContent(jsonBuilder(), ToXContent.EMPTY_PARAMS));
+
+                    clientWithOrigin.index(
+                        indexRequest,
+                        ActionListener.wrap(
+                            indexResponse -> listener.onResponse(new PostConnectorSyncJobAction.Response(indexResponse.getId())),
+                            listener::onFailure
+                        )
+                    );
+                } catch (IOException e) {
+                    listener.onFailure(e);
+                }
+            }));
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    private String generateId() {
```

Comment:
(Assuming the answer to your TODO is "no it's not needed") It looks like `DocWriteResponse` has a `getId()` method, could we use that for the response and get rid of this?